py-rse/ci.Rmd

```diff
@@ -108,168 +94,146 @@ knitr::include_graphics("figures/rse-ci/add-repo.png")
-(which most people look at more often).
-To add a status badge,
-click the "build icon" shown in the top right corner of Figure \@ref(fig:image-github-icon)
-to bring up a dialog box.
-Select "Markdown" from the "Format" menu list,
-then copy the Markdown text displayed in the "Result" box
-and paste it into the project's `README.md` file.
-(It's best to paste the text right below the page's title so that it will be instantly visible).
-Once we commit and push thsi change to GitHub,
-our project will now show the badge for its Travis CI status Figure \@ref(fig:image-github-icon).
-
-```{r py-rse-ci-github-icon, echo=FALSE, fig.cap="The Travis CI Build Badge"}
-knitr::include_graphics("figures/rse-ci/github-icon.png")
-```
+### Adding Travis CI build commands
 
-> **How Do Badges Work?**
->
-> Badges are much simpler than they first appear.
-> When someone visits our project,
-> GitHub sends them the HTML of the project's home page.
-> That page includes a link to an image on Travis CI's site,
-> so our browser sends a request to Travis CI.
-> The URL tells Travis CI's web server what project we're interested in;
-> it then sends back a different image
-> depending on the most recently recorded build status for that project.
+After having added the `.travis.yml` file to the repository,
+any changes pushed to the GitHub repo will trigger Travis to run its commands.
+You may have noticed that Travis didn't really do anything to the code in your project repository.
+That's because we haven't yet told Travis to do anything!
+Given that we've created some unit tests from Chapter \@ref(py-rse-correct),
+let's get Travis to run those tests on its build system.
+In the `.travis.yml` file, add a new line and write out:
 
-## How can I set up Travis CI for an R project? {#py-rse-ci-setting-up-with-r}
+```yaml
+language: python
 
-The simplest way to set up and use Travis CI with R is to call:
+python:
+- "3.6"
 
-```r
-usethis::use_travis()
+script:
+- pytest
 ```
 
-This command:
-
--   Creates a `.travis.yml` configuration file with typical settings for an R package.
--   Adds a Travis CI build badge to the project's `README.md`.
--   Opens the Travis CI page for the package so that we can activate Travis CI for the repository.
-    (This assumes the package is already on GitHub and that we have a Travis CI account.)
-
-FIXME: add example here of generated YAML file
+Here we've added a setting called `script`
+that tells Travis to run the following list of commands,
+in this case, only `pytest`.
+We can put almost anything here as long as it doesn't need human interaction
+(i.e., doesn't ask questions).
 
-The [usethis](https://usethis.r-lib.org/reference/ci.html) documentation has more details,
-but it really is this simple.
+Getting CI to run the project tests is the main reason to use CI,
+so that any change made can be tested for correctness.
+And because Travis runs in a "clean" environment (aka, not your own computer),
```

Comment:
Why the quotes around "clean" in this context?

Reply:
Clarified. "clean" cause it was never "dirty" :wink: